### PR TITLE
Update to io-ts version 2

### PR DESCRIPTION
**Pull request by artfwo**

### CODE REVIEW

File: src/types/Union.ts [hunk 1]
```diff
@@ -2,11 +2,16 @@ import * as t from 'io-ts'
...
 
-export const UnionOf: UnionOfType = <T extends string>(values: T[], name?: string): Union<T> => {
-    return t.union(
-        values.map(val => t.literal(val)),
-        name ? `Union:${name}` : 'Union'
-    )
+
+
+export const UnionOf: UnionOfType = <T extends string>(values: [T, T, ...T[]], name?: string): Union<T> => {
+    const [ value1, value2, ...valuesRest ] = values
```
Comment (milos-mircov-fh) [inline src/types/Union.ts]: Can you please explain why this change `value1, value2, ...valuesRest`?

Comment (artfwo) [reply src/types/Union.ts]: UnionType in io-ts expects a tuple with at least 2 values, so here we're using same type signature and have to destructure it accordingly.
